inc/commoninjectionlib.class.php

```diff
@@ -1532,9 +1532,16 @@ private function effectiveAddOrUpdate($injectionClass, $item, $values, $add = tr
             $toinject[$key] = $value;
          }
 
-         //useful for Infocom
-         if (get_class($item) == Infocom::getType() &&
-         ($key === 'items_id' || $key === 'itemtype')) {
+         if (
```

Comment:
I do not excatly how works this plugin, but maybe a more generic way could be used to preserve relation fields.
Is there any case where it would be a problem to preserve `items_id` and `itemtype` ?

Reply:
I'm not sure i want to pass another couple of hours on the issue.
This plugin is a pain to debug.